modules/persistent-storage-csi-manila-dynamic-provisioning.adoc

```diff
@@ -0,0 +1,87 @@
+// Module included in the following assemblies:
+//
+// * storage/container_storage_interface/persistent-storage-csi-manila.adoc
+
+[id="persistent-storage-csi-manila-dynamic-provisioning_{context}"]
+= Dynamically provisioning Manila CSI volumes
+
+{product-title} installs a default StorageClass.
+
+The YAML files that are created are completely decoupled from Manila and from its Container Storage Interface (CSI) plug-in. As an application developer, you can dynamically provision ReadWriteMany (RWX) storage and deploy Pods with applications that safely consume the storage using YAML manifests.
+
+You can use the same Pod and persistent volume claim (PVC) definitions on-premise that you use with {product-title} on AWS, GCP, Azure, and other platforms, with the exception of the storage class reference in the PVC definition.
+
+.Prerequisites
+
+* OpenStack is deployed with appropriate Manila share infrastructure so that it can be used to dynamically provision and mount volumes in {product-title}.
+
+.Procedure (UI)
+
+To dynamically create a Manila CSI volume using the web console:
+
+. In the {product-title} console, click *Storage* → *Persistent Volume Claims*.
+
+. In the persistent volume claims overview, click *Create Persistent Volume Claim*.
+
+. Define the required options on the resulting page.
+
+.. Select the appropriate StorageClass.
+
+.. Enter a unique name for the storage claim.
+
+.. Select the access mode to specify read and write access for the PVC you are creating.
++
+[IMPORTANT]
+====
+* Use RWX if you want the persistent volume (PV) that fulfills this PVC to be mounted to multiple Pods on multiple nodes in the cluster.
+
+* Use RWO mode if you want to prevent additional Pods from being dynamically provisioned.
+====
+
+. Define the size of the storage claim.
+
+. Click *Create* to create the PersistentVolumeClaim and generate a PersistentVolume.
+
+.Procedure (CLI)
+
+To dynamically create a Manila CSI volume using the command-line interface (CLI):
+
+. Create and save a file with the PersistentVolumeClaim object described by the following YAML:
+
++
+.pvc-manila.yaml
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-manila
+spec:
+  accessModes: <1>
+    - ReadWriteMany
+  resources:
+    requests:
+  	storage: 10Gi
+  storageClassName: gold <2>
```

Comment:
All StorageClasses for Manila are provisioned by the Operator and have the `csi-manila-` prefix: https://github.com/openshift/csi-driver-manila-operator/blob/master/pkg/controller/maniladriver/manila_storageclasses.go#L17